File: ios/LunarCore/Tests/LunarCoreTests/RandomDateTests.swift
//
//  RandomDateTests.swift
//  LunarCoreTests
//
//  Random date conversion and round-trip.
//

import Testing
import LunarCore

@Suite("Random date conversion")
struct RandomDateTests {

    @Test("Round-trip solar -> lunar -> solar")
    func roundTrip() {
        let engine = LunarEngine()
        let dates: [(Int, Int, Int)] = [
            (1990, 5, 15),
            (2000, 12, 31),
            (1985, 3, 8),
            (2010, 7, 20),
        ]
        for (y, m, d) in dates {
            let solar = SolarDate(year: y, month: m, day: d)
            let lunar = engine.solarToLunar(date: solar)
            let back = engine.lunarToSolar(date: lunar)
            #expect(back.year == solar.year)
            #expect(back.month == solar.month)
            #expect(back.day == solar.day)
        }
    }

    @Test("Lunar day in range 1-30")
    func lunarDayRange() {
        let engine = LunarEngine()
        let solar = SolarDate(year: 2020, month: 6, day: 15)
        let lunar = engine.solarToLunar(date: solar)
        #expect(lunar.day >= 1 && lunar.day <= 30)
        #expect(lunar.month >= 1 && lunar.month <= 12)
    }
}

File: android/lunarcore/src/test/kotlin/com/lunarcore/RandomDateTests.kt
package com.lunarcore

import kotlin.test.Test
import kotlin.test.assertTrue

class RandomDateTests {

    @Test
    fun roundTrip() {
        val engine = LunarEngine()
        val dates = listOf(
            Triple(1990, 5, 15),
            Triple(2000, 12, 31),
            Triple(1985, 3, 8),
            Triple(2010, 7, 20)
        )
        for ((y, m, d) in dates) {
            val solar = SolarDate(y, m, d)
            val lunar = engine.solarToLunar(solar)
            val back = engine.lunarToSolar(lunar)
            assertTrue(back.year == solar.year && back.month == solar.month && back.day == solar.day)
        }
    }

    @Test
    fun lunarDayInRange() {
        val engine = LunarEngine()
        val solar = SolarDate(2020, 6, 15)
        val lunar = engine.solarToLunar(solar)
        assertTrue(lunar.day in 1..30)
        assertTrue(lunar.month in 1..12)
    }
}
